hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMCertStore.java

```diff
@@ -122,19 +150,16 @@ public X509Certificate getCertificateByID(BigInteger serialID,
   public List<X509Certificate> listCertificate(HddsProtos.NodeType role,
       BigInteger startSerialID, int count, CertType certType)
       throws IOException {
-    // TODO: Filter by role
     List<? extends Table.KeyValue<BigInteger, X509Certificate>> certs;
+
     if (startSerialID.longValue() == 0) {
       startSerialID = null;
     }
-    if (certType == CertType.VALID_CERTS) {
-      certs = scmMetadataStore.getValidCertsTable().getRangeKVs(
-          startSerialID, count);
-    } else {
-      certs = scmMetadataStore.getRevokedCertsTable().getRangeKVs(
-          startSerialID, count);
-    }
+
+    certs = getCertTableList(role, certType, startSerialID, count);
+
     List<X509Certificate> results = new ArrayList<>(certs.size());
+
     for (Table.KeyValue<BigInteger, X509Certificate> kv : certs) {
       try {
         X509Certificate cert = kv.getValue();
```

Comment:
The TODO: comment below can be removed as we are using different tables to do the filtering.

Reply:
Removed it.